### ISSUE TITLE
load_jit.go error terminate called after throwing an instance of 'c10::Error'  

### ISSUE THREAD

**Issue by gujinmal** [kuroko1t/gotorch#14]
I have a problem when I ran load_jit.go

ERROR
terminate called after throwing an instance of 'c10::Error'
  what():  [enforce fail at inline_container.cc:222] . file not found: archive/constants.pkl
frame #0: c10::ThrowEnforceNotMet(char const*, int, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, void const*) + 0x67 (0x7f93c27aa6c7 in /home/pin/IdeaProjects/Go_Server_FaceDetection/src/github.com/kuroko1t/gotorch/libtorch/lib/libc10.so)
frame #1: caffe2::serialize::PyTorchStreamReader::getRecordID(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0xd6 (0x7f93b2cef926 in /home/pin/IdeaProjects/Go_Server_FaceDetection/src/github.com/kuroko1t/gotorch/libtorch/lib/libtorch_cpu.so)
frame #2: caffe2::serialize::PyTorchStreamReader::getRecord(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0x34 (0x7f93b2cef9a4 in /home/pin/IdeaProjects/Go_Server_FaceDetection/src/github.com/kuroko1t/gotorch/libtorch/lib/libtorch_cpu.so)
frame #3: torch::jit::readArchiveAndTensors(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, c10::optional<std::function<c10::StrongTypePtr (c10::QualifiedName const&)> >, c10::optional<std::function<c10::intrusive_ptr<c10::ivalue::Object, c10::detail::intrusive_target_default_null_type<c10::ivalue::Object> > (c10::StrongTypePtr, c10::IValue)> >, c10::optional<c10::Device>, caffe2::serialize::PyTorchStreamReader&) + 0x97 (0x7f93b42e3117 in /home/pin/IdeaProjects/Go_Server_FaceDetection/src/github.com/kuroko1t/gotorch/libtorch/lib/libtorch_cpu.so)
frame #4: <unknown function> + 0x36c5790 (0x7f93b42e3790 in /home/pin/IdeaProjects/Go_Server_FaceDetection/src/github.com/kuroko1t/gotorch/libtorch/lib/libtorch_cpu.so)
frame #5: <unknown function> + 0x36c8302 (0x7f93b42e6302 in /home/pin/IdeaProjects/Go_Server_FaceDetection/src/github.com/kuroko1t/gotorch/libtorch/lib/libtorch_cpu.so)
frame #6: torch::jit::load(std::shared_ptr<caffe2::serialize::ReadAdapterInterface>, c10::optional<c10::Device>, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >&) + 0x1a4 (0x7f93b42e6cd4 in /home/pin/IdeaProjects/Go_Server_FaceDetection/src/github.com/kuroko1t/gotorch/libtorch/lib/libtorch_cpu.so)
frame #7: torch::jit::load(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, c10::optional<c10::Device>, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >&) + 0xc5 (0x7f93b42e9a85 in /home/pin/IdeaProjects/Go_Server_FaceDetection/src/github.com/kuroko1t/gotorch/libtorch/lib/libtorch_cpu.so)
frame #8: torch::jit::load(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, c10::optional<c10::Device>) + 0x6f (0x7f93b42e9b8f in /home/pin/IdeaProjects/Go_Server_FaceDetection/src/github.com/kuroko1t/gotorch/libtorch/lib/libtorch_cpu.so)
frame #9: Load + 0x74 (0x67e4b4 in /tmp/go-build157121112/b001/exe/load_model)
frame #10: _cgo_0aee684483a6_Cfunc_Load + 0x1b (0x66715b in /tmp/go-build157121112/b001/exe/load_model)
frame #11: /tmp/go-build157121112/b001/exe/load_model() [0x481270]

**Comment by kuroko1t** [comment]
According to [this issue](https://github.com/pytorch/pytorch/issues/47917), this error seems to occur when loading a pytorch model with libtorch. Please check [the linked issue](https://github.com/pytorch/pytorch/issues/47917) for details. 
It is assumed to load a model in torchscript format, so check the format of the model.

**Comment by jhillhouse92** [comment]
Right, JIT load is not the same as python `torch.load`. JIT load is only loading a TorchScript model export which is either from torch.jit.trace (traced) or torch.jit.script (scripted). The trace is likely to be more performant but requires exact example input with no control flow in module whereas script module is more flexible support dynamic inputs and control flow. This go package with the signature of `torch.Load` is actually the python `torch.jit.load`. See https://pytorch.org/docs/stable/jit.html for more info.

**Comment by jhillhouse92** [comment]
@kuroko1t I think the action item here would be to rename the method to `JITLoad` to add more clarity. I'm relatively new to Go but it doesn't seem like one can do `Torch.JIT.Load`.
